src/cmd/services/m3coordinator/ingest/write.go

```diff
@@ -446,6 +461,7 @@ func (d *downsamplerAndWriter) writeAggregatedBatch(
 				},
 			}
 		}
+		opts.MetricType = value.Attributes.Type
```

Comment:
nit: Instead of setting this here can you set it in the original declaration?
```go
opts := downsample.SampleAppenderOptions{
  MetricsType: value.Attributes.Type,
}
if downsampleMappingRuleOverrides, ok := d.downsampleOverrideRules(overrides); ok {
  // ...
}
```